test/components/connect.spec.js

```diff
@@ -1117,6 +1117,45 @@ describe('React', () => {
       expect(decorated.refs.wrappedInstance.someInstanceMethod()).toBe(someData);
     });
 
+    it('should not add refs to wrapped instance if stateless', () => {
+      const store = createStore(() => ({}));
+      const decorator = connect(state => state);
+      const Decorated = decorator(() => <div/>);
+
+      const tree = TestUtils.renderIntoDocument(
+        <ProviderMock store={store}>
+          <Decorated />
+        </ProviderMock>
+      );
+
+      const decorated = TestUtils.findRenderedComponentWithType(tree, Decorated);
+      expect(Object.keys(decorated.refs)).toExclude('wrappedInstance');
+    });
+
+    it('should consider module pattern of non-Component-inheriting classes stateful', () => {
+      function MyComponent(initialProps) {
+        return {
+          state: { value: initialProps.initialValue },
+          render: function render() {
+            return <span className={this.state.value} />;
+          }
+        };
+      }
```

Comment:
Hmm, interestingly, this turns out to not be a false positive. At least, React thinks this is stateless, and shows the warning.